Core/Utilities/Subprocess.py

```diff
@@ -368,6 +370,10 @@ def __readFromFile( self, fd, baseLength, doAll = True ):
         if nB == "":
           break
         dataString += nB
+        #break out of potential infinite loop, indicated by dataString growing beyond reason
+        if len( dataString ) > datastringMaxAllowedLength:
```

Comment:
Why not to compare as
if len( dataString ) + baseLength > self.bufferLimit:
as this will be checked later